### ISSUE TITLE
Documentation: Mypy-friendly non-dynamic explicit base example is misleading

### ISSUE THREAD

**Issue by posita** [sqlalchemy/sqlalchemy#7575]
### Describe the bug

From [this](https://docs.sqlalchemy.org/en/14/orm/mapping_styles.html#creating-an-explicit-base-non-dynamically-for-use-with-mypy-similar) [[source](https://github.com/sqlalchemy/sqlalchemy/blob/master/doc/build/orm/mapping_styles.rst#creating-an-explicit-base-non-dynamically-for-use-with-mypy-similar)]:

``` python
from sqlalchemy.orm import registry
from sqlalchemy.orm.decl_api import DeclarativeMeta

mapper_registry = registry()


class Base(metaclass=DeclarativeMeta):
    __abstract__ = True

    # these are supplied by the sqlalchemy2-stubs, so may be omitted
    # when they are installed
    registry = mapper_registry
    metadata = mapper_registry.metadata

    __init__ = mapper_registry.constructor


print(Base.registry)
print(Base.metadata)
```

What does, "these are supplied by the sqlalchemy2-stubs, so may be omitted when they are installed," mean (from [this example](https://docs.sqlalchemy.org/en/14/orm/mapping_styles.html#creating-an-explicit-base-non-dynamically-for-use-with-mypy-similar))? What does the "these" refer to? The next two lines or four? What does the "they" refer to? The stubs?

### To Reproduce

```python
In any case, those lines seem important, even if `sqlalchemy2-stubs` is installed (which contradicts my best reading of that vague comment):

 python
# test_case.py
from sqlalchemy.orm import registry
from sqlalchemy.orm.decl_api import DeclarativeMeta

mapper_registry = registry()


class Base(metaclass=DeclarativeMeta):
    __abstract__ = True
    __init__ = mapper_registry.constructor


print(Base.registry)
print(Base.metadata)
```

### Error

Mypy doesn't complain, but the runtime environment sure does:

``` sh
% pip list | grep -i alchemy
SQLAlchemy        1.4.29
sqlalchemy2-stubs 0.0.2a19
% python test_case2.py
Traceback (most recent call last):
  File "/…/test_case.py", line 8, in <module>
    class Base(metaclass=DeclarativeMeta):
  File "/…/lib/python3.9/site-packages/sqlalchemy/orm/decl_api.py", line 64, in __init__
    raise exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: Declarative base class has no 'registry' attribute, or registry is not a sqlalchemy.orm.registry() object
```

The comment in the original example makes it sounds like some of those lines are optional and one can remove them if one has `sqlalchemy2-stubs` installed. That does not appear true.

### Versions

- Python: 3.9.9
- SQLAlchemy: 1.4.29 (sqlalchemy2-stubs 0.0.2a19)
- Database: N/A
- DBAPI (eg: psycopg, cx_oracle, mysqlclient): N/A

**Comment by zzzeek** [comment]
> What does, "these are supplied by the sqlalchemy2-stubs, so may be omitted when they are installed," mean (from [this example](https://docs.sqlalchemy.org/en/14/orm/mapping_styles.html#creating-an-explicit-base-non-dynamically-for-use-with-mypy-similar))? 

it means the two attributes "registry" and "metadata", as publicly accessible attributes, will be known to the type checker because these are defined by sqlalchemy2-stubs as elements of the DeclarativeMeta metaclass, at least at some point in the development of the stubs package.

> What does the "these" refer to? The next two lines or four? 

the next two lines

> What does the "they" refer to? The stubs?

the stubs

> In any case, those lines seem important, even if `sqlalchemy2-stubs` is installed (which contradicts my best reading of that vague comment):

at some point something must have changed, and this comment should be removed from the documentation, the two lines in question when using this particular recipe should always be included.

The whole system will be replaced by #7562 in 2.0.

PRs welcome to remove the two lines from the example documentation.

**Comment by posita** [comment]
It looks like this is fixed in `main` at 01c50c64e302c193733cef7fb146fbab8eaa44bd (fixing #7257). (I didn't realize, since `master` still exists, which is confusing in its own right.) Assuming `main` is the branch from which the next 1.4 release will occur, I think this can be closed? If there's another branch to which the change needs to be propagated, let me know, and I'll submit a PR.

**Comment by zzzeek** [comment]
fixed for 1.4 in [f688a81](https://github.com/sqlalchemy/sqlalchemy/commit/f688a81c479a3b5fa0d355ce17ffa58c363ae0d6)
